internal/repository/targets_test.go

```diff
@@ -47,9 +47,15 @@ func TestAddDelegation(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	gpgKey, err := gpg.LoadGPGKeyFromBytes(gpgPubKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	targetsMetadata, err := state.GetTargetsMetadata(policy.TargetsRoleName)
 	assert.Nil(t, err)
-	assert.Empty(t, targetsMetadata.Delegations.Keys)
+	assert.Contains(t, targetsMetadata.Delegations.Keys, gpgKey.KeyID)
+	assert.Equal(t, 1, len(targetsMetadata.Delegations.Keys))
```

Comment:
Let's instead assert there are 2 rules, as we're adding one more in this test and we assert again it was added.

Reply:
I have moved the check and added an assert.Equal for the length of the keys

Reply:
We still need to assert the len(targetsMetadata.Delegations.Roles) here so that we see it's 2, then 3 after AddDelegation.

Reply:
You correctly do the inverse of the check for Remove below, 3 then 2.

Reply:
Sorry for the late reply, I have fixed this.